src/client.rs

```diff
@@ -313,8 +313,10 @@ where
 
         let mut target_pool: ConnectionPool = ConnectionPool::default();
         let mut transaction_mode = false;
+        let mut server_info = BytesMut::default();
```

Comment:
I get a warning here because `BytesMut::default()` value was overwritten before it was used. 
Open for ideas on how to refactor this to avoid the warning.

Reply:
You can make if blocks return values, e.g.

```rust
let value = if true {
   5
}
else
{
   6
}
```

so you don't have to have a default value up one scope level like in C/C++.